### DISCUSSION TITLE
COPY CON - multiple lines and typed characters to be visible

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#4052] [General]
VirtualBox with MS-DOS 6.22:
![image](https://github.com/user-attachments/assets/cd2b039a-8bef-4372-801d-32fcd8c44e03)

Staging 0.82:
![image](https://github.com/user-attachments/assets/f7066f6b-7c92-4bbf-b587-5c493b32ebb9)

What's missing in Staging:
- multiple lines - currently when you press Enter it saves the file
- while you type to see the characters on screen - currently they're written correctly to the file at the end, but not visible while typing
- F6 to behave like Ctrl+Z - currently it adds ` @` to the file

Note: currently Staging shows an extra "CON" when saving (which MS-DOS doesn't show) and that's a useful improvement.

[Extra info](https://www.os2museum.com/wp/misconceptions-on-top-of-misconceptions/) on Ctrl+Z, F6, copy con

Related: [file creation date not registered](https://github.com/dosbox-staging/dosbox-staging/discussions/4139) when using COPY CON in mounted IMG.

**Comment by johnnovak** [comment]
Extremely edge-casey and we don't really care about this, IMO. 

People who rely on `copy con` and want _exact_ MS-DOS behaviour should get a life 😎

This isn't necessary to _play games_, and normal people just use a text editor 🤷🏻
